src/regdesk/login/login_manager.js

```diff
@@ -8,6 +8,45 @@ const STORAGE_LOCAL_KEY = 'regfox-bearer-details';
  * 1. Keeps the bearer token fresh and regfox status connected.
  */
 export class LoginManager {
+  static #instance;
+  /** Only one instance of LoginManager can exist. */
+  constructor() {
+    if (LoginManager.#instance) {
+      throw new Error('LoginManager already has an instance!!!');
+    }
+    LoginManager.#instance = this;
+
+    setInterval(async () => {
+      try {
+        await this.refresh();
+      } catch (e) {
+        console.error('Failed to refresh bearer token, user will need to re-login.', e);
+        await this.#setBearerDetails(null);
```

Comment:
I wonder if we should add an event of some sort here that we could use to update the connected state? Or should this directly manage the "Connected" status in the top left?

Reply:
Haha, thats my next pull request!